framework/include/utils/MooseTypes.h

```diff
@@ -166,6 +166,8 @@ enum KernelType
 {
   KT_TIME = 0,
   KT_NONTIME = 1,
+  KT_NONEIGEN = 2,
+  KT_EIGEN  = 3,
```

Comment:
`KT_TIME` and `KT_NONTIME` is used by time integration system. The constants _have_ to stay, because they are used in `Assembly` to index into local residuals for time and non-time contribution for the residual. They are not used only for kernel type...